### ISSUE TITLE
Add support for MAIL FROM domain

### ISSUE THREAD

**Issue by john-tipper** [binxio/cfn-ses-provider#12]
`cfn-ses-provider` does not currently support custom MAIL FROM domains: https://docs.aws.amazon.com/ses/latest/DeveloperGuide/mail-from.html

Thanks for all the work you've put in to these, and the accompanying blog posts, they're extremely useful. I'm happy to contribute a patch to this if you'd like and think it worthwhile. Would you be happy with the call to [`SetIdentityMailFromDomain`](https://docs.aws.amazon.com/ses/latest/APIReference/API_SetIdentityMailFromDomain.html) being inside `get_token()` within `domain_identity_provider.py`?

**Comment by mvanholsteijn** [comment]
Hi John, I looked at the call, and I think it is best implemented using its own custom domain `Custom::MailFromDomain`, As you can only set it after the domain is specified. I also think it allows you to specify multiple subdomains. 

Will you make a PR for it?

**Comment by john-tipper** [comment]
Hi Mark,

Yes, happy to look at doing a PR, but a bit pushed for time this week, so sorry for the delay. I'm struggling to get the tests to run though, probably something to do with my AWS setup (I think it's perhaps some prerequisites not being satisfied).

I assume that to run the tests I should do `make test`, but my tests are failing at `test_identity_policy_provider.py`:

```
../tests/test_active_rule_set_provider.py ......                                                                                                                        [ 37%]
../tests/test_dkim_tokens_provider.py .                                                                                                                                 [ 43%]
../tests/test_domain_identity_provider.py .                                                                                                                            [ 50%]
../tests/test_identity_notifications_provider.py ....                                                                                                                 [ 75%]
../tests/test_identity_policy_provider.py F                                                                                                                              [ 81%]

```

I get the following message:

```
 def test_create():
        identity = f"{uuid.uuid4()}.internal"
        account_id = (boto3.client("sts")).get_caller_identity()["Account"]
        region = "eu-west-1"
    
        try:
            request = Request("Create", identity, account_id, region)
            response = handler(request, {})
>           assert response["Status"] == "SUCCESS", response["Reason"]
E           AssertionError: could not set domain identity policy MyPolicy, An error occurred (InvalidParameterValue) when calling the PutIdentityPolicy operation: Invalid identity. Must be a verified email address or domain.
E           assert 'FAILED' == 'SUCCESS'
E             - SUCCESS
E             + FAILED

../tests/test_identity_policy_provider.py:15: AssertionError

```

I'm not sure whether there is any setup required, other than a usable AWS profile? I'm running the tests with AWS_REGION and AWS_DEFAULT_REGION set to `eu-west-1`. Is there something else I need to do to get the tests to run correctly please?

**Comment by mvanholsteijn** [comment]
Hi John,

If you  put in your PR, I will check it out; The tests are not unit tests:
they are integration tests so it may be that it won't work in your
environment.

Cheers,

Mark

**Comment by john-tipper** [comment]
Hi Mark,

I've added an initial PR #13 for support for a custom MAIL FROM domain and also a custom resource for waiting until it becomes verified - I haven't attempted to add integration tests as these are for your Binx internal network but as the code draws very heavily on the existing `DomainIdentity` and `VerifiedIdentity` resources hopefully there should be relatively little work involved to incorporate these additional tests. I tend to do relatively little with Python, so if you encounter any issues or have any comments on the code then I'm very happy to receive feedback. 

Many thanks,

John

**Comment by mvanholsteijn** [comment]
Solved in release 0.7.0. Thank you @john-tipper !
